include/DeterministicNoiseSimulator.hpp

```diff
@@ -3,124 +3,132 @@
 
 #include "QuantumComputation.hpp"
 #include "Simulator.hpp"
+#include "dd/NoiseFunctionality.hpp"
 
 #include <cstddef>
 #include <map>
 #include <memory>
+#include <optional>
 #include <random>
 #include <string>
 #include <vector>
 
-class DeterministicNoiseSimulator: public Simulator<dd::DensityMatrixSimulatorDDPackage> {
-    using CN    = dd::ComplexNumbers;
+template<class DDPackage = DensityMatrixPackage>
+class DeterministicNoiseSimulator: public Simulator<DDPackage> {
     using dEdge = dd::dEdge;
+    using dNode = dd::dNode;
 
 public:
-    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, unsigned long long seed):
-        Simulator(seed), qc(qc) {
-    }
-
-    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, const std::string& noise_effects, double noise_prob):
-        qc(qc) {
-        setNoiseEffects(noise_effects);
-        initializeNoiseProbabilities(noise_prob);
-    }
-
-    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, const std::string& noise_effects, double noise_prob, double amp_noise_prob, double multiQubitGateFactor):
-        qc(qc) {
-        setNoiseEffects(noise_effects);
-        initializeNoiseProbabilities(noise_prob, amp_noise_prob, multiQubitGateFactor);
-    }
-
-    double noiseProb                 = 0.0;
-    double ampDampingProb            = 0.0;
-    double noiseProbSingleQubit      = 0.0;
-    double ampDampingProbSingleQubit = 0.0;
-    double noiseProbMultiQubit       = 0.0;
-    double ampDampingProbMultiQubit  = 0.0;
+    explicit DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, unsigned long long seed = 0):
+        DeterministicNoiseSimulator(qc, std::string("APD"), 0.001, std::optional<double>{}, 2, false, seed) {}
+
+    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc,
+                                const std::string&                       cNoiseEffects,
+                                double                                   cNoiseProbability,
+                                std::optional<double>                    cAmpDampingProbability,
+                                double                                   cMultiQubitGateFactor,
```

Comment:
no need to have a preceding `c` in the parameter names

Reply:
Changed that.